modules/afsocket/afsocket-dest.c

```diff
@@ -428,6 +428,7 @@ afsocket_dd_try_connect(AFSocketDestDriver *self)
       afsocket_dd_start_reconnect_timer(self);
       return;
     }
+  g_assert(!iv_timer_registered(&self->reconnect_timer));
   self->reconnect_timer.handler = (void (*)(void *)) afsocket_dd_reconnect;
```

Comment:
I don't think it is good practice to change the handler of a timer during execution.

I'd either keep state and react accordingly in the same function _OR_ use a separate timer for the two cases.

Reply:
Since the assertion that reconnect timer is not running fails in some cases (thanks @kira-syslogng for finding it), I'll refactor the `reconnect_timer`'s handler changing one too.

Reply:
I've removed the handler-changing during execution. It's part of a massive refactor.